finagle-mysql/src/main/scala/com/twitter/finagle/mysql/Parameter.scala

```diff
@@ -0,0 +1,37 @@
+package com.twitter.finagle.exp.mysql
+
+import com.twitter.finagle.exp.mysql.transport.BufferWriter
+import language.implicitConversions
+
+trait Parameter {
+  type A
+  def value: A
+  def evidence: CanBeParameter[A]
+
+  final def writeTo(writer: BufferWriter): Unit = {
+    evidence.write(writer, value)
+  }
+
+  final def size = evidence.sizeOf(value)
+  final def typeCode = evidence.typeCode(value)
+}
+
+object Parameter {
+  implicit def wrap[_A](_value: _A)(implicit _evidence: CanBeParameter[_A]): Parameter = {
```

Comment:
@travisbrown I am not sure there is a precedent for either convention in Scala, since it's not a very widely used pattern. But I agree that `A0` might be a better convention that `_A`. Will change.